internal/bzlmod/go_mod.bzl

```diff
@@ -240,16 +322,37 @@ def sums_from_go_mod(module_ctx, go_mod_label):
     """
     _check_go_mod_name(go_mod_label.name)
 
-    # We go through a Label so that the module extension is restarted if go.sum
+    return parse_sumfile(module_ctx, go_mod_label, "go.sum")
+
+def sums_from_go_work(module_ctx, go_work_label):
```

Comment:
I believe the existence of a `go.work.sum`  signals that a go.mod is lacking a `go mod tidy` invocation. We could be strict, and error. Or we could warn or we could do nothing...

For correctness, we likely want to error, and I'm leaving towards this. But I'm curious what go engineers would like in practice